drivers/sensor/lis2dux12/Kconfig

```diff
@@ -0,0 +1,29 @@
+# ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+
+# Copyright (c) 2023 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig LIS2DUX12
+	bool "LIS2DUX12 I2C/SPI accelerometer sensor driver"
+	default y
+	depends on DT_HAS_ST_LIS2DUX12_ENABLED
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_LIS2DUX12),i2c)
+	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_LIS2DUX12),spi)
+	select HAS_STMEMSC
+	select USE_STDC_LIS2DUX12
```

Comment:
So this needs the hal_st repo yet doesn't actually care to check if it exists and drivers themselves select that the module is present, well that's one way to note do it. This needs to be completely reworked for this driver and for every other driver in tree, this should be checking if the module is present i.e. `depends on ZEPHYR_HAL_ST_MODULE` if it requires that module be present, otherwise I can get example_application, build that sample for a board but enable a sensor and get build failures because it's including that are not present because the module is not enabled

Reply:
@erwango not sure if this is an ST employee or not but if not then please ping someone that is to take note of this comment

Reply:
I'm not am ST employee and this is my first Zephyr contribution. This driver was based on another driver with a nearly identical `Kconfig`, including a recently merged (https://github.com/zephyrproject-rtos/zephyr/commit/1badec4bfda81707ed406bef82fedfa0250a59f0). I've only been using Zephyr for a couple months so I'm not aware of the subtleties of the build system as you're pointing out.

I'm happy to make the necessary changes, though I'm afraid it may require a bit more explicit direction.

Reply:
This is something ST need to fix, if drivers require a hal being available then the options should not be selectable if the hal is not available because doing so would lead to a build error and it doesn't make sense to be able to select something that can't actually be used.

Reply:
^^@avisconti

Reply:
> So this needs the hal_st repo yet doesn't actually care to check if it exists and drivers themselves select that the module is present, well that's one way to not do it. This needs to be completely reworked for this driver and for every other driver in tree, this should be checking if the module is present i.e. `depends on ZEPHYR_HAL_ST_MODULE` if it requires that module be present, otherwise I can get example_application, build that sample for a board but enable a sensor and get build failures because it's including that are not present because the module is not enabled

@nordicjm let me get this straight, because I'm not sure I see the point. So, you are requesting that we should avoid the use of ```select HST_STMEMSC``` and ```select USE_STDC_LIS2DUX12``` and instead just put a ```depends on ZEPHYR_HAL_ST_MODULE``` (or whatever) in this file. Correct? But here the intention is in fact to indeed enable the specific STDC driver inside HAL_STM32 module. What am I not getting?

Reply:
> If you have an in-tree driver that requires the hal_st repo be present to compile then those drivers should have a `depends on ZEPHYR_HAL_ST_MODULE` or similar so that they can only be selected if the module is present. If they build fine without the repo being present then it is not needed - see #67242 for an example

But is it not required to run ```west update``` before building the code to get the system matching what the manifest looks like?

Reply:
Not sure what that has to do with it? A user having zephyr does not mean they have hal_st, the whole example_application repo for example will not get hal_st https://github.com/zephyrproject-rtos/example-application/blob/main/west.yml

Reply:
> Not sure what that has to do with it? A user having zephyr does not mean they have hal_st, the whole example_application repo for example will not get hal_st https://github.com/zephyrproject-rtos/example-application/blob/main/west.yml

got it, sorry

Reply:
@avisconti done!